enforcer/src/keystate/keystate_ds_submit_task.c

```diff
@@ -48,7 +48,7 @@ keystate_ds_submit_task_perform(task_type *task)
 		KEY_DATA_DS_AT_PARENT_SUBMIT, KEY_DATA_DS_AT_PARENT_SUBMITTED,
 		(engine_type*)task->context);
 	task_cleanup(task);
-	flush_enforce_task((engine_type*)task->context, 0);
+	flush_enforce_task((engine_type*)task->context, time_now());
```

Comment:
That does not do what you think it does.
int flush_enforce_task(engine_type *engine, bool enforce_all);
Also, on schedule_flush_type() which is called by the above function. the time is set to time_now() rather than 0. So it already does what you want.